### ISSUE TITLE
Kics reports an error on Ansible commented values

### ISSUE THREAD

**Issue by LvffY** [Checkmarx/kics#4563]
This follows the issue #4419 that has been marked as resolved by #4503. It seems that the issue is still real and more tedious than expected.

### Expected Behavior

**Kics** should not report that a he finds a password in my file.

### Actual Behavior

Beside my value is just an example and is commented, **kics** reports an error

### Steps to Reproduce the Problem

1.  Have the following **group_vars/backends/vars.yml**

```yaml
---
backends:
  db_name: "db"
  user:
    username: "user"
#    definition of using host_vars variable in an other variable
#    hostvars:
#      airflow_1:
#        vault_demo_airflow_user_1_password: value
    password: "{{ another_encrypted_value }}"
```

2.  In `checkmarx/kics:debian` run the analysis `/app/bin/kics scan -p group_vars/backends/vars.yml -o ${PWD} --report-formats json,sarif --ignore-on-exit results --verbose --log-level=DEBUG`

3.  [Obtain this kind of output ](https://github.com/Checkmarx/kics/files/7603039/results.txt)

***N.B:*** I tried multiple ignore method explained from the [official documentation](https://docs.kics.io/latest/running-kics/#using_commands_on_scanned_files_as_comments) but none worked. I tried the following : 

- Completely ignore file ❌
```yaml
---
# kics-scan ignore
backends:
  db_name: "db"
  user:
    username: "user"
#    definition of using host_vars variable in an other variable
#    hostvars:
#      airflow_1:
#        vault_demo_airflow_user_1_password: value
    password: "{{ another_encrypted_value }}"
```
- Ignore block ❌
```yaml
---
backends:
  db_name: "db"
  user:
    username: "user"
#    definition of using host_vars variable in an other variable
#    hostvars:
#      airflow_1:
# kics-scan ignore-block
#        vault_demo_airflow_user_1_password: value
    password: "{{ another_encrypted_value }}"
```
- Ignore line ❌
```yaml
---
backends:
  db_name: "db"
  user:
    username: "user"
#    definition of using host_vars variable in an other variable
#    hostvars:
#      airflow_1:
# kics-scan ignore-line
#        vault_demo_airflow_user_1_password: value
    password: "{{ another_encrypted_value }}"
```

### Specifications

-   Version: 1.4.8
-   Platform: Ansible
-   Subsystem: N/A

**Comment by rafaela-soares** [comment]
Hi, @LvffY ! Thank you so much for reporting the bug again. We apologize.

In #4503, we added the consideration of LinesIgnore in passwords and secrets inspector. However, the bug resides specifically on the Ignores Lines mechanism for YAML files. We will fix it as soon as possible. 

Until then, you can use `# kics-scan ignore` on the top of the file (first line):

```
# kics-scan ignore
---
backends:
  db_name: "db"
  user:
    username: "user"
#    definition of using host_vars variable in an other variable
#    hostvars:
#      airflow_1:
#        vault_demo_airflow_user_1_password: value
    password: "{{ another_encrypted_value }}"
```

**Comment by rafaela-soares** [comment]
Hello again, @LvffY 🙂

For now, you can also use `# kics-scan ignore-line` or `# kics-scan ignore-block` without commenting on the target line/block:
```
---
backends:
  db_name: "db"
  user:
    username: "user"
#    definition of using host_vars variable in an other variable
    hostvars:
      airflow_1:
         # kics-scan ignore-line
         vault_demo_airflow_user_1_password: value
    password: "{{ another_encrypted_value }}"
```

**Comment by LvffY** [comment]
Hi @rafaela-soares 

Hope you're doing well.

No problem about reopening the issue, don't worry. It was supposed to be fixed with the `kics-scan ignore-line` feature and I just needed some time to test it :)

About your workaround, I'm not sure about your idea. Because my commenting line is meant to be commented. Actually, in my use case, I have a very complex variable to generate. This variable is generated programmatically through Ansible templating. Because of this complexity, I added some comments about the real value generated. 

To be more precise, my real YAML file is the following one : 

```yaml
---
# backends hosts look like spark, airflow_1 and value should look like demo_spark_user and demo_airflow_user_1
# backend_name get the name of the host, here spark and airflow
backend_name: "{{ (inventory_hostname.split('_')[:-1] | join('_')) if ((inventory_hostname.split('_') | length) > 2) else inventory_hostname.split('_')[0] }}"
# bakend_number get the number of the host, here "" and _1
backend_number: "{{ '_' + inventory_hostname.split('_')[-1] if ((inventory_hostname.split('_') | length) > 1) else '' }}"

backends:
  db_name: "demo_{{ backend_name }}_db{{ backend_number }}"
  user:
    username: "demo_{{ backend_name }}_user{{ backend_number }}"
    #    definition of using host_vars variable in an other variable
    #    hostvars:
    #      airflow_1:
    #        vault_demo_airflow_user_1_password: value
    password: "{{ hostvars[inventory_hostname]['vault_demo_'+ backend_name + '_user' + backend_number + '_password' ] }}"
```

So as you can see, the calculation to obtain my field `password` is *really* not trivial. Because of this I wanted to add some comments to explicit the real value.

So in your workaround, my comment is not a comment anymore et will be interpreted by Ansible as a real value. I don't want this because it could lead to Ansible inventory mess up :(

Please tell me if anything is not clear

BTW, this is not really hurry for me, for now we're POCing you're solution and we can admit some known issues :)

Have a great day and merry christmas to all of you ;)

![image](https://user-images.githubusercontent.com/37327656/147355494-c21ec1fb-0958-4b68-8609-e03a794189d1.png)

**Comment by rafaela-soares** [comment]
Hello, @LvffY 🙂 

I am doing well, thank you. I hope this message finds you well too. And I hope you had a wonderful Christmas! 


Thank you so much for your message and patience! We are very happy to inform you that we start to fix it in PR #4662. It should finally resolve it! 

About the workaround, it was just a suggestion to avoid the KICS results. But not a great one, since you needed to comment on your lines again after running your project against KICS. So, for now, I think my first suggestion (`# kics-scan ignore`) should be the one that fits better for you.

I take this opportunity to wish you a good year 2022 🎉 🎉 🎉 🎉 

![adf34244b76162cb152f8af7cf525b19](https://user-images.githubusercontent.com/74001161/148102856-b824ee4b-1ae8-47f9-84be-6f96801b4fad.gif)
